synapse/storage/databases/main/purge_events.py

```diff
@@ -249,12 +249,11 @@ def _purge_history_txn(
         # Mark all state and own events as outliers
         logger.info("[purge] marking remaining events as outliers")
         txn.execute(
-            "UPDATE events SET outlier = ?"
-            " WHERE event_id IN ("
-            "    SELECT event_id FROM events_to_purge "
-            "    WHERE NOT should_delete"
-            ")",
-            (True,),
+            "UPDATE events SET outlier = TRUE"
+            "WHERE event_id IN ("
```

Comment:
Looks like the space was lost here but it's important as all of these strings just get concatted together

```suggestion
            " WHERE event_id IN ("
```


CI failure: https://github.com/matrix-org/synapse/actions/runs/5478999712/jobs/9982296723?pr=15525#step:6:941
```
  File "/synapse/build/debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.11/site-packages/synapse/storage/databases/main/purge_events.py", line 251, in _purge_history_txn
    txn.execute(
  File "/synapse/build/debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.11/site-packages/synapse/storage/database.py", line 417, in execute
    self._do_execute(self.txn.execute, sql, parameters)
  File "/synapse/build/debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.11/site-packages/synapse/storage/database.py", line 469, in _do_execute
    return func(sql, *args, **kwargs)
sqlite3.OperationalError: near "event_id": syntax error
```